docs/DeveloperGuide.md

```diff
@@ -1,29 +1,136 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started](#setting-up-getting-started)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#UI-Component)
+    * [Parser Component](#parser-component)
+    * [BankWithUs Component](#bankwithus-component)
+    * [Account Component](#account-component)
+    * [AccountList Component](#accountlist-component)
+    * [SaveGoal Component](#saveGoal-component)
+    * [Storage Component](#storage-component)
+    * [Withdrawal Limit Component](#withdraw limit checker-Component)
+    * [Transaction Component](#transaction-Component)
+    * [TransactionList Component](#transactionlist-component)
+* [Implementation](#implementation)
+    * [Account related Feature](#Account-feature)
+    * [saveGoal related Feature](#SaveGoal-feature)
+    * [transaction Related Feature](#transaction-feature)
+
+* [Appendix: Requirements](#appendix-requirements)
+    * [Product scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+    * [User Stories](#user-stories)
+    * [Non-Functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+
+---
+
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Setting up, getting started
+
+---
+
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Design
 
-## Product scope
-### Target user profile
+### Architecture
+<img src="images/MainArchitecture.png" width="280" />
 
-{Describe the target user profile}
+**Interactions**
 
-### Value proposition
+The sequence diagram below shows how components interact with each other when the user issues the command `deposit 100`.
+
+<img src="images/deposit100SeqDiagram.png" width="280" />
+
+
+### UI-Component
+Class: `Ui.java`
+
+*  contains all code that interfaces directly with the CLI
+
+### Parser-Component
+Class: `Parser.java`
+
+* determines what the program would do with the input retrieved from the CLI
+
+### Account-Component
+Class: `Account.java`
```

Comment:
The class descriptions are not uniformly formatted (some start with "The ___ component:", some don't, some start with capital letters, some don't, some don't have a description (Withdrawal-Limit-Checker-Component), I think it would be beneficial to make all the class descriptions uniformly formatted to increase readability and understandability for other developers